### ISSUE TITLE
Bug: "HTTP Error. Upload URL might be wrong or doesn't exist." my runtime is silverlight

### ISSUE THREAD

**Issue by techno489** [moxiecode/plupload#1084]
<p><br>
I have been trying to get this to upload large files 1GB in size<br>
I have read many many posts and tried numbers of options but simply can't get 
large files to upload, I can upload 300MB files but any more then that I get 
&quot;HTTP Error. Upload URL might be wrong or doesn't exist.&quot;<br>
<br>
<b><u>my jquery_ui_widget.html code is</u></b><br>
<br>
&lt;!DOCTYPE html&gt;<br>
&lt;html xmlns=&quot;http://www.w3.org/1999/xhtml&quot; dir=&quot;ltr&quot;&gt;<br>
&lt;head&gt;<br>
&lt;meta http-equiv=&quot;content-type&quot; content=&quot;text/html; charset=UTF-8&quot;/&gt;<br>
<br>
&lt;title&gt;Plupload - jQuery UI Widget&lt;/title&gt;<br>
<br>
&lt;link rel=&quot;stylesheet&quot; href=&quot;//ajax.googleapis.com/ajax/libs/jqueryui/1.8.9/themes/base/jquery-ui.css&quot; 
type=&quot;text/css&quot; /&gt;<br>
&lt;link rel=&quot;stylesheet&quot; href=&quot;../../js/jquery.ui.plupload/css/jquery.ui.plupload.css&quot; 
type=&quot;text/css&quot; /&gt;<br>
<br>
&lt;script src=&quot;//ajax.googleapis.com/ajax/libs/jquery/1.9.0/jquery.min.js&quot;&gt;&lt;/script&gt;<br>
&lt;script type=&quot;text/javascript&quot; src=&quot;//ajax.googleapis.com/ajax/libs/jqueryui/1.10.2/jquery-ui.min.js&quot;&gt;&lt;/script&gt;<br>
<br>
&lt;!-- production --&gt;<br>
&lt;script type=&quot;text/javascript&quot; src=&quot;../../js/plupload.full.min.js&quot;&gt;&lt;/script&gt;<br>
&lt;script type=&quot;text/javascript&quot; src=&quot;../../js/jquery.ui.plupload/jquery.ui.plupload.js&quot;&gt;&lt;/script&gt;<br>
<br>
&lt;!-- debug <br>
&lt;script type=&quot;text/javascript&quot; src=&quot;../../js/moxie.js&quot;&gt;&lt;/script&gt;<br>
&lt;script type=&quot;text/javascript&quot; src=&quot;../../js/plupload.dev.js&quot;&gt;&lt;/script&gt;<br>
&lt;script type=&quot;text/javascript&quot; src=&quot;../../js/jquery.ui.plupload/jquery.ui.plupload.js&quot;&gt;&lt;/script&gt;<br>
--&gt;<br>
<br>
&lt;/head&gt;<br>
&lt;body style=&quot;font: 13px Verdana; background: #eee; color: #333&quot;&gt;<br>
<br>
&lt;h1&gt;jQuery UI Widget&lt;/h1&gt;<br>
<br>
&lt;p&gt;You can see this example with different themes on the &lt;a href=&quot;http://plupload.com/example_jquery_ui.php&quot;&gt;www.plupload.com&lt;/a&gt; 
website.&lt;/p&gt;<br>
<br>
&lt;form id=&quot;form&quot; method=&quot;post&quot; action=&quot;../dump.php&quot;&gt;<br>
&lt;div id=&quot;uploader&quot;&gt;<br>
&lt;p&gt;Your browser doesn't have Flash, Silverlight or HTML5 support.&lt;/p&gt;<br>
&lt;/div&gt;<br>
&lt;br /&gt;<br>
&lt;input type=&quot;submit&quot; value=&quot;Submit&quot; /&gt;<br>
&lt;/form&gt;<br>
<br>
&lt;script type=&quot;text/javascript&quot;&gt;<br>
// Initialize the widget when the DOM is ready<br>
$(function() {<br>
$(&quot;#uploader&quot;).plupload({<br>
// General settings<br>
// runtimes : 'html5,flash,silverlight,html4',<br>
runtimes : 'silverlight',<br>
url : '../upload.php',<br>
<br>
// User can upload no more then 20 files in one go (sets multiple_queues to 
false)<br>
max_file_count: 20,<br>
<br>
chunk_size: '1mb',<br>
<br>
// Resize images on clientside if we can<br>
resize : {<br>
width : 200, <br>
height : 200, <br>
quality : 90,<br>
crop: true // crop to exact dimensions<br>
},<br>
<br>
filters : {<br>
// Maximum file size<br>
max_file_size : '1000mb',<br>
// Specify what files to browse for<br>
mime_types: [<br>
{title : &quot;Video files&quot;, extensions : 
&quot;mpg,mpeg,divx,avi,mp4,swf,wmv,xvid,yuv,vob&quot;},<br>
{title : &quot;Audio files&quot;, extensions : &quot;aac,wav,mp3&quot;},<br>
{title : &quot;Image files&quot;, extensions : &quot;jpg,gif,png,bmp&quot;},<br>
{title : &quot;Zip files&quot;, extensions : &quot;7-Zip,zip,rar&quot;}<br>
]<br>
},<br>
<br>
// Rename files by clicking on their titles<br>
rename: true,<br>
<br>
// Sort files<br>
sortable: true,<br>
<br>
// Enable ability to drag'n'drop files onto the widget (currently only HTML5 
supports that)<br>
dragdrop: true,<br>
<br>
// Views to activate<br>
views: {<br>
list: true,<br>
thumbs: true, // Show thumbs<br>
active: 'thumbs'<br>
},<br>
<br>
// Flash settings<br>
flash_swf_url : '../../js/Moxie.swf',<br>
<br>
// Silverlight settings<br>
silverlight_xap_url : '../../js/Moxie.xap'<br>
});<br>
<br>
<br>
// Handle the case when form was submitted before uploading has finished<br>
$('#form').submit(function(e) {<br>
// Files in queue upload them first<br>
if ($('#uploader').plupload('getFiles').length &gt; 0) {<br>
<br>
// When all files are uploaded submit form<br>
$('#uploader').on('complete', function() {<br>
$('#form')[0].submit();<br>
});<br>
<br>
$('#uploader').plupload('start');<br>
} else {<br>
alert(&quot;You must have at least one file in the queue.&quot;);<br>
}<br>
return false; // Keep the form from submitting<br>
});<br>
});<br>
&lt;/script&gt;<br>
&lt;/body&gt;<br>
&lt;/html&gt;<br>
<br>
<br>
&lt;/code&gt;<br>
<br>
<br>
<b><u>my upload.php code is this</u></b><br>
&lt;code&gt;<br>
&lt;?php<br>
/**<br>
* upload.php<br>
*<br>
* Copyright 2013, Moxiecode Systems AB<br>
* Released under GPL License.<br>
*<br>
* License: http://www.plupload.com/license<br>
* Contributing: http://www.plupload.com/contributing<br>
*/<br>
<br>
#!! IMPORTANT: <br>
#!! this file is just an example, it doesn't incorporate any security checks and
<br>
#!! is not recommended to be used in production environment as it is. Be sure to
<br>
#!! revise it and customize to your needs.<br>
<br>
<br>
// Make sure file is not cached (as it happens for example on iOS devices)<br>
header(&quot;Expires: Mon, 26 Jul 1997 05:00:00 GMT&quot;);<br>
header(&quot;Last-Modified: &quot; . gmdate(&quot;D, d M Y H:i:s&quot;) . &quot; GMT&quot;);<br>
header(&quot;Cache-Control: no-store, no-cache, must-revalidate&quot;);<br>
header(&quot;Cache-Control: post-check=0, pre-check=0&quot;, false);<br>
header(&quot;Pragma: no-cache&quot;);<br>
<br>
/* <br>
// Support CORS<br>
header(&quot;Access-Control-Allow-Origin: *&quot;);<br>
// other CORS headers if any...<br>
if ($_SERVER['REQUEST_METHOD'] == 'OPTIONS') {<br>
exit; // finish preflight CORS requests here<br>
}<br>
*/<br>
<br>
// 5 minutes execution time<br>
// @set_time_limit(5 * 60);<br>
<br>
// Uncomment this one to fake upload time<br>
// usleep(5000);<br>
<br>
// Settings<br>
$targetDir = ini_get(&quot;upload_tmp_dir&quot;) . DIRECTORY_SEPARATOR . &quot;plupload&quot;;<br>
$targetDir = 'uploads';<br>
// $cleanupTargetDir = fauls; // Remove old files<br>
// $maxFileAge = 5 * 3600; // Temp file age in seconds<br>
<br>
<br>
// Create target dir<br>
if (!file_exists($targetDir)) {<br>
@mkdir($targetDir);<br>
}<br>
<br>
// Get a file name<br>
if (isset($_REQUEST[&quot;name&quot;])) {<br>
$fileName = $_REQUEST[&quot;name&quot;];<br>
} elseif (!empty($_FILES)) {<br>
$fileName = $_FILES[&quot;file&quot;][&quot;name&quot;];<br>
} else {<br>
$fileName = uniqid(&quot;file_&quot;);<br>
}<br>
<br>
$filePath = $targetDir . DIRECTORY_SEPARATOR . $fileName;<br>
<br>
// Chunking might be enabled<br>
$chunk = isset($_REQUEST[&quot;chunk&quot;]) ? intval($_REQUEST[&quot;chunk&quot;]) : 0;<br>
$chunks = isset($_REQUEST[&quot;chunks&quot;]) ? intval($_REQUEST[&quot;chunks&quot;]) : 0;<br>
<br>
<br>
// Remove old temp files <br>
if ($cleanupTargetDir) {<br>
if (!is_dir($targetDir) || !$dir = opendir($targetDir)) {<br>
die('{&quot;jsonrpc&quot; : &quot;2.0&quot;, &quot;error&quot; : {&quot;code&quot;: 100, &quot;message&quot;: &quot;Failed to open temp 
directory.&quot;}, &quot;id&quot; : &quot;id&quot;}');<br>
}<br>
<br>
while (($file = readdir($dir)) !== false) {<br>
$tmpfilePath = $targetDir . DIRECTORY_SEPARATOR . $file;<br>
<br>
// If temp file is current file proceed to the next<br>
if ($tmpfilePath == &quot;{$filePath}.part&quot;) {<br>
continue;<br>
}<br>
<br>
// Remove temp file if it is older than the max age and is not the current file<br>
if (preg_match('/\.part$/', $file) &amp;&amp; (filemtime($tmpfilePath) &lt; time() - $maxFileAge)) 
{<br>
@unlink($tmpfilePath);<br>
}<br>
}<br>
closedir($dir);<br>
} <br>
<br>
<br>
// Open temp file<br>
if (!$out = @fopen(&quot;{$filePath}.part&quot;, $chunks ? &quot;ab&quot; : &quot;wb&quot;)) {<br>
die('{&quot;jsonrpc&quot; : &quot;2.0&quot;, &quot;error&quot; : {&quot;code&quot;: 102, &quot;message&quot;: &quot;Failed to open 
output stream.&quot;}, &quot;id&quot; : &quot;id&quot;}');<br>
}<br>
<br>
if (!empty($_FILES)) {<br>
if ($_FILES[&quot;file&quot;][&quot;error&quot;] || !is_uploaded_file($_FILES[&quot;file&quot;][&quot;tmp_name&quot;])) 
{<br>
die('{&quot;jsonrpc&quot; : &quot;2.0&quot;, &quot;error&quot; : {&quot;code&quot;: 103, &quot;message&quot;: &quot;Failed to move 
uploaded file.&quot;}, &quot;id&quot; : &quot;id&quot;}');<br>
}<br>
<br>
// Read binary input stream and append it to temp file<br>
if (!$in = @fopen($_FILES[&quot;file&quot;][&quot;tmp_name&quot;], &quot;rb&quot;)) {<br>
die('{&quot;jsonrpc&quot; : &quot;2.0&quot;, &quot;error&quot; : {&quot;code&quot;: 101, &quot;message&quot;: &quot;Failed to open 
input stream.&quot;}, &quot;id&quot; : &quot;id&quot;}');<br>
}<br>
} else { <br>
if (!$in = @fopen(&quot;php://input&quot;, &quot;rb&quot;)) {<br>
die('{&quot;jsonrpc&quot; : &quot;2.0&quot;, &quot;error&quot; : {&quot;code&quot;: 101, &quot;message&quot;: &quot;Failed to open 
input stream.&quot;}, &quot;id&quot; : &quot;id&quot;}');<br>
}<br>
}<br>
<br>
while ($buff = fread($in, 4096)) {<br>
fwrite($out, $buff);<br>
}<br>
<br>
@fclose($out);<br>
@fclose($in);<br>
<br>
// Check if file has been uploaded<br>
if (!$chunks || $chunk == $chunks - 1) {<br>
// Strip the temp .part suffix off <br>
rename(&quot;{$filePath}.part&quot;, $filePath);<br>
}<br>
<br>
// Return Success JSON-RPC response<br>
die('{&quot;jsonrpc&quot; : &quot;2.0&quot;, &quot;result&quot; : null, &quot;id&quot; : &quot;id&quot;}');<br>
&lt;/code&gt;<br>
<br>
I think I need something to keep the connection to the domain active but not 
sure.<br>
I only know very little about coding and mostly try what I read others have 
done.<br>
<br>
the closest topic I found was https://github.com/moxiecode/plupload/issues/855<br>
but that didn't help me or I didn't fully understand how he fixed his problem<br>
<br>
I edited the php.ini on my domain &quot;max_execution_time = 0&quot; as I read somewhere 
to do this to stop drop out with lard files but it still didn't fix my problem.<br>
<br>
Any help with this would be great, I have been busy with this now for about 5 
days spending most my time trying to make this work.<br>
<br>
Cheers</p>

**Comment by jaydip321** [comment]
Hi jayarjo,
How did you overcome this issue ?
i also have same kind of issue. i am uploading 100 images at once with each of 2 MB. It gives same error after uploading 15 images. after skipping about 20 images it again uploads next 15 images.

i changed server settings also.

please help me if you overcame this issue.
Thanks

**Comment by kushlces** [comment]
I am currently having this issue. For me, I don't think its related to the server configuration but rather a netwrok issue. If I change my ISP, it works. Basically, file gets uploaded with some internet providers and not with others. I will update if I figure out the exact cause.

**Comment by kjin9174** [comment]
Hello jayarjo
are you round?

**Comment by kjin9174** [comment]
I wanna know how url of upload.js works?

**Comment by kjin9174** [comment]
how to connect upload.js with upload.php???

**Comment by jayarjo** [comment]
@kjin9174 we have a getting started guide here: http://www.plupload.com/docs/v2/Getting-Started

**Comment by haiderpro** [comment]
This is an old thread but this problem occurred to our server today and after investing a good amount of time, I found that the problem was due to Apache's mod_evasive module. If the following error is haunting you:
"Upload URL might be wrong or doesn't exist"

Then ask your web host to check if Apache's mod_evasive module is installed on the server and ask them to disable or remove it and the problem will be fixed.

Apache's mod_evasive module is to avoid brute force attack but it false positively detect file uploads as attack and blocks the file uploads. There may be other reasons also but if some files are uploading and some are not then this is what is going to solve the issue.

Sharing this information to make things easier for others :-)
